File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/simple/common/vehicle/Truck.java
package com.java.se.conclusion.design.factory.simple.common.vehicle;

/**
 * 	This is a class to be used to test the implementation of simple factory pattern
 *  -- This class must inherit a parent abstract class
 *  
 * @author VinceYuan
 *
 */
public class Truck extends Vehicle {

	@Override
	public void turn() {
		System.out.println("A truck is turning left...");
	}

	@Override
	public void inspect() {
		System.out.println("A truck is under inspection...");
	}
}

File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/simple/common/vehicle/Truck.scala
package org.lang.scala.conclusion.design.factory.simple.common.vehicle

import org.lang.scala.conclusion.design.factory.simple.common.vehicle.parent.Vehicle

/**
 * 	This is a stand-alone class to be used to test the implementation of simple factory pattern
 *  -- This class must inherit a parent abstract class
 *  
 * 	@author VinceYuan
 */
class Truck extends Vehicle {
  
	override def turn(): Unit = {
		println("A truck is turning left...")
	}

	override def inspect(): Unit = {
		println("A truck is under inspection...")
	}
}